xbout/boutdataarray.py

```diff
@@ -133,6 +135,243 @@ def fromFieldAligned(self):
         result["direction_y"] = "Standard"
         return result
 
+    def fromRegion(self, region, with_guards=None):
+        """
+        Get a logically-rectangular section of data from a certain region.
+        Includes guard cells from neighbouring regions.
+
+        Parameters
+        ----------
+        region : str
+            Region to get data for
+        with_guards : int or dict of int, optional
+            Number of guard cells to include, by default use MXG and MYG from BOUT++.
+            Pass a dict to set different numbers for different coordinates.
+        """
+
+        region = self.data.regions[region]
+        xcoord = self.data.metadata['bout_xdim']
+        ycoord = self.data.metadata['bout_ydim']
+        keep_yboundaries = self.data.metadata['keep_yboundaries']
+
+        if with_guards is None:
+            mxg = self.data.metadata['MXG']
+            myg = self.data.metadata['MYG']
+        else:
+            try:
+                try:
+                    mxg = with_guards[xcoord]
+                except KeyError:
+                    mxg = self.data.metadata['MXG']
```

Comment:
Just a note that this is equivalent to `mxg = with_guards.get(xcoord, default=self.data.metadata["MXG"])` :)